### PR TITLE
Removes 'Ion' naming convention in order to align with naming convention of ion-js

### LINKED ISSUE #25: Remove 'Ion' prefix from API
Names without the 'Ion' prefix are expected to be clear enough without the prefix, and use of this library is expected to be relatively localized (thus minimal likelihood of conflicting with a name from another library, and in the event of a conflict, an alias can be used).  Or usage of these names will already be prefixed with a module name (e.g., `ion.HashReader`, whereas `ion.IonHashReader` is redundant).

Doing so would also bring this library's naming convention closer to that followed by ion-js.

### PR DESCRIPTION
**Pull request by pbcornell**
Resolves #25 

Build succeeds against the HEAD of ion-js.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/IonHash.ts [hunk 1]
```diff
@@ -13,85 +13,84 @@
...
-import {Reader as IonReader} from 'ion-js';
-import {Writer as IonWriter} from 'ion-js';
+import {Reader, Writer} from 'ion-js';
 
-import {_HashReaderImpl, _HashWriterImpl, _CryptoIonHasher} from './internal/IonHashImpl';
+import {_HashReaderImpl, _HashWriterImpl, _CryptoHasher} from './internal/IonHashImpl';
 
 /**
- * Wraps the provided IonReader as an IonHashReader.
+ * Wraps the provided Reader as an HashReader.
```
Comment (zslayton) [inline src/IonHash.ts]: nit: `as an HashReader` -> `as a HashReader`

File: src/IonHash.ts [hunk 2]
```diff
@@ -13,85 +13,84 @@
...
-export function makeHashReader(reader: IonReader,
-                               hashFunctionProvider: IonHasherProvider): IonHashReader {
+export function makeHashReader(reader: Reader,
+                               hashFunctionProvider: HasherProvider): HashReader {
     return new _HashReaderImpl(reader, hashFunctionProvider);
 }
 
 /**
- * Wraps the provided IonWriter as an IonHashWriter.
+ * Wraps the provided Writer as an HashWriter.
```
Comment (zslayton) [inline src/IonHash.ts]: `an HashWriter` -> `a HashWriter`

File: src/IonHash.ts [hunk 3]
```diff
@@ -13,85 +13,84 @@
...
-     * @throws if invoked at a different depth than when the IonHashWriter was instantiated
+     * @throws if invoked at a different depth than when the HashWriter was instantiated
      */
     digest(): Uint8Array;
 }
 
 
 /**
- * Implementations of this function type interface create an IonHasher
+ * Implementations of this function type interface create an Hasher
```
Comment (zslayton) [inline src/IonHash.ts]: `an Hasher` -> `a Hasher`

File: src/IonHash.ts [hunk 4]
```diff
@@ -109,14 +108,14 @@ export interface IonHasher {
 }
 
 /**
- * An IonHasherProvider implementation backed by node's [crypto](https://node.readthedocs.io/en/latest/api/crypto/)
+ * An HasherProvider implementation backed by node's [crypto](https://node.readthedocs.io/en/latest/api/crypto/)
```
Comment (zslayton) [inline src/IonHash.ts]: `An Hasher` -> `A Hasher`

File: src/IonHash.ts (hunk 1 above)
Comment (pbcornell) [reply src/IonHash.ts]: good catch, fixed, thanks

File: src/IonHash.ts (hunk 2 above)
Comment (pbcornell) [reply src/IonHash.ts]: good catch, fixed, thanks

File: src/IonHash.ts (hunk 3 above)
Comment (pbcornell) [reply src/IonHash.ts]: good catch, fixed, thanks

File: src/IonHash.ts (hunk 4 above)
Comment (pbcornell) [reply src/IonHash.ts]: good catch, fixed, thanks

**Pull request merged**